src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1259,25 +1167,24 @@ private void requestTool(@NotNull final Block target)
      * @param tool     the tool needed
      * @param required the level needed (for pickaxe only)
      */
-    private void updateToolFlag(@NotNull final String tool, final int required)
+    private void updateToolFlag(@NotNull final ToolType toolType, final int required)
     {
-        switch (tool)
+        switch (toolType)
         {
-            case Utils.AXE:
-                checkForAxe();
-                break;
-            case Utils.SHOVEL:
-                checkForShovel();
+            case AXE:
+            case SHOVEL:
+            case HOE:
+            case BOW:
+            case SWORD:
+                checkForTool(toolType);
                 break;
-            case Utils.HOE:
-                checkForHoe();
-                break;
-            case Utils.PICKAXE:
-                checkForPickaxe(required);
+            case PICKAXE:
+                checkForTool(toolType, required);
                 break;
             default:
-                checkForPickaxe(DIAMOND_LEVEL);
-                Log.getLogger().error("Invalid tool " + tool + " not implemented as tool will require pickaxe level 4 instead.");
+                //TODO checkForTool(TOOL_LEVEL_DIAMOND);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)